mat64/io_test.go

```diff
@@ -0,0 +1,42 @@
+// Copyright ©2015 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package mat64
+
+import (
+	"bytes"
+	"encoding/gob"
+	"testing"
+)
+
+func TestDenseRW(t *testing.T) {
+	for i, test := range []Matrix{
+		NewDense(0, 0, []float64{}),
+		NewDense(2, 2, []float64{1, 2, 3, 4}),
+		NewDense(2, 3, []float64{1, 2, 3, 4, 5, 6}),
+		NewDense(3, 2, []float64{1, 2, 3, 4, 5, 6}),
+		NewDense(3, 3, []float64{1, 2, 3, 4, 5, 6, 7, 8, 9}),
+		NewDense(3, 3, []float64{1, 2, 3, 4, 5, 6, 7, 8, 9}).View(0, 0, 2, 2),
+		NewDense(3, 3, []float64{1, 2, 3, 4, 5, 6, 7, 8, 9}).View(1, 1, 2, 2),
+		NewDense(3, 3, []float64{1, 2, 3, 4, 5, 6, 7, 8, 9}).View(0, 1, 3, 2),
+	} {
+		buf := new(bytes.Buffer)
+		enc := gob.NewEncoder(buf)
+		err := enc.Encode(test)
+		if err != nil {
+			t.Fatalf("error encoding test #%d: %v\n", i, err)
+		}
```

Comment:
Could you please add a test that the returned bytes is the correct length? It should be rows*cols*8 + 16 bytes.